orttraining/orttraining/test/gradient/function_ops_test.cc

```diff
@@ -36,244 +21,44 @@ static void RegisterSchemas() {
-    input_values.push_back(std::make_pair(input_name, ort_value));
-    input_value_map.insert(std::make_pair(input_name, ort_value));
-  }
-
-  void AddOutput(std::string output_name) {
-    output_names.emplace_back(output_name);
-    output_args.emplace_back(output_name, nullptr);
-  }
-
-  void AddAttribute(const char* attr_name, int64_t attr_val) {
-    ONNX_NAMESPACE::AttributeProto axis_attr;
-    axis_attr.set_name(attr_name);
-    axis_attr.set_type(ONNX_NAMESPACE::AttributeProto_AttributeType::AttributeProto_AttributeType_INT);
-    axis_attr.set_i(attr_val);
-    attributes[attr_name] = axis_attr;
-  }
-
-  onnxruntime::Node& AddCallNodeTo(onnxruntime::Graph& graph) {
-    std::vector<NodeArg*> input_arg_ptrs;
-
-    for (auto& arg : input_args)
-      input_arg_ptrs.push_back(&arg);
-
-    std::vector<NodeArg*> output_arg_ptrs;
-    for (auto& arg : output_args)
-      output_arg_ptrs.push_back(&arg);
-
-    return graph.AddNode("fncallnode", opname, "function call node", input_arg_ptrs, output_arg_ptrs, &attributes, onnxruntime::kMSDomain);
-  }
-
-  std::unique_ptr<Model> CreateModel(bool inline_call = false) {
-    RegisterSchemas();
-    if (opsets.size() == 0) {
-      // Default opsets
-      opsets[kOnnxDomain] = 13;
-      opsets[kMSDomain] = 1;
-    }
-
-    std::unique_ptr<Model> model(new Model("test", false, ModelMetaData(), PathString(), IOnnxRuntimeOpSchemaRegistryList(),
-                                           opsets, {}, DefaultLoggingManager().DefaultLogger()));
-
-    onnxruntime::Graph& graph = model->MainGraph();
-    auto& call_node = AddCallNodeTo(graph);
-
-    auto status = graph.Resolve();
-    EXPECT_TRUE(status.IsOK()) << status.ErrorMessage();
-
-    if (inline_call) {
-      graph.InlineFunction(call_node);
-#if _LOCAL_DEBUG_FLAG_
-      std::cout << graph << std::endl;
-#endif
-      status = graph.Resolve();
-      EXPECT_TRUE(status.IsOK()) << status.ErrorMessage();
-    }
-
-    return model;
-  }
-
-  void RunTest() {
-    auto model1 = CreateModel(false);
-    auto results1 = Run(*model1, input_value_map, output_names);
-
-    auto model2 = CreateModel(true);
-    auto results2 = Run(*model2, input_value_map, output_names);
-
-    AssertEqual(results1, results2);
-  }
-};
-
 static void InitSoftmaxGradTestCase(FunctionTestCase& testCase, std::vector<int64_t> shape) {
   testCase.AddInput<float>("dY", shape);
   testCase.AddInput<float>("Y", shape);
   testCase.AddOutput("dX");
 }
 
 TEST(SoftmaxGradExpansionTest, DefaultAxis) {
+  RegisterSchemas();
   FunctionTestCase testCase("SoftmaxGrad");
   InitSoftmaxGradTestCase(testCase, {3, 2});
   testCase.RunTest();
 }
 
 TEST(SoftmaxGradExpansionTest, NegativeAxis) {
+  RegisterSchemas();
   FunctionTestCase testCase("SoftmaxGrad");
   InitSoftmaxGradTestCase(testCase, {3, 2});
   testCase.AddAttribute("axis", -1);
   testCase.RunTest();
 }
 
 TEST(SoftmaxGradExpansionTest, PositiveAxis) {
+  RegisterSchemas();
```

Comment:
>RegisterSchemas [](http://example.com/codeflow?start=2&length=15)

Seems repeating. Could we use fixtures to setup in one class? [https://github.com/google/googletest/blob/master/docs/primer.md#test-fixtures-using-the-same-data-configuration-for-multiple-tests-same-data-multiple-tests](https://github.com/google/googletest/blob/master/docs/primer.md#test-fixtures-using-the-same-data-configuration-for-multiple-tests-same-data-multiple-tests)

Reply:
Done